adrs/01-AccessControlPolicies.md

```diff
@@ -0,0 +1,82 @@
+
+### Overview of ShinzoHub's role
+
+To facilitate these ACPs, ShinzoHub will take a very important role.
+
+First, Shinzo will leverage a niche document creation flow exposed by SourceHub where, when a document is created, ownership rights are immediately relinquised to the SourceHub protocol itself. In SourceHub, the owner of a document has complete control over that document; they may read, update, delete, etc. the document as they please. While this behaviour is desirable for most DefraDB applications which emphasize giving users (and other entities) control over their own data, Shinzo's purpose is instead to give the community better control of all blockchain data. For this reason, transferring control to the protocol itself aligns more closely with Shinzo's vision.
+
+ShinzoHub will be given a special role on all documents and groups used by Shinzo, the "admin" role. This role gives ShinzoHub the permission to manage sync, write, and read permissions on each document on a did or group basis and also gives ShinzoHub the permission to add and remove people from the Indexer and Host groups. In practice, this gives ShinzoHub 3 main responsibilities:
+
+1) ShinzoHub needs to assign the appropriate read/write permissions for Indexer and Host groups on each document created at the time of creation (once admin rights are assigned). This assignment of permissions is done either via a direct assignment of roles to the groups or via establishing a parent relationship with the collection the data has been derived from (see Views section).
+
+2) ShinzoHub needs to expose APIs where prospective Indexers and Hosts can register to join their respective group. This registration process may require the collection, validation, and/or storage of some data.
+
+3) `read` permissions on a Primitive or View will require payment. While the details are not yet finalized, payments are likely to be facilitated by a smart contract deployed on each Shinzo-supported chain; the Shinzo team has nicknamed this the "Outpost" contract for now. The Outpost contract will emit events whenever a successful payment is made. ShinzoHub will be required to listen for these events via IBC and then will need to give the paying did the appropriate role(s) such that they will have `read` permissions on the chosen Primitive or View.
+
+### Node Access Control
+
+By default, when working with SourceHub and Defra, any node has the ability to create new policies and attach relationships to them. This kind of flexible access control, while valuable for some use cases, is not desirable for Shinzo - Shinzo's goal is to make blockchain data highly available to paying users via a decentralized protocol and infrastructure, not to give user's ownership over data. In the case of Shinzo, the protocol, ShinzoHub, owns the data and it alone should maintain administrator priveledges on the data.
```

Comment:
Probably doesn't matter too much in case of Shinzo, but for accuracy sake here are some comments:

> By default

For DefraDB, currently they need to opt into it by starting a node like for example:
```sh
defradb start --no-keyring --node-acp-enable true --identity e3b722906ee4e56368f581cd8b18ab0f48af1ea53e635e3f7b8acd076676f6ac
```

> any node has the ability to create new policies

Currently for NAC this is not possible. An internal policy designed by Defra is uploaded manually at the moment.